File: articles/analytical-psychology-and-literary-interpretation/fortran/symbolic_density.f90
! Analytical Psychology and Literary Interpretation
! Fortran example: simple symbolic-density score

program symbolic_density_demo
  implicit none

  integer :: motif_count, total_tokens
  real :: density

  motif_count = 8
  total_tokens = 120

  if (total_tokens > 0) then
     density = real(motif_count) / real(total_tokens)
  else
     density = 0.0
  end if

  print *, "Synthetic symbolic-density score:", density
end program symbolic_density_demo

File: articles/analytical-psychology-and-literary-interpretation/rust/symbolic_density.rs
fn symbolic_density(motif_count: u32, total_tokens: u32) -> f64 {
    if total_tokens == 0 {
        return 0.0;
    }
    motif_count as f64 / total_tokens as f64
}

fn main() {
    let score = symbolic_density(8, 120);
    println!("Synthetic symbolic-density score: {:.4}", score);
}
